src/node.hpp

```diff
@@ -50,12 +50,12 @@ class Node {
     std::unordered_map<std::string, Aliases> blobNamesMapping;
 
     // TODO make fields below const after integration of PipelineDefinition with Demultiplexer/Gather
-    std::optional<std::set<std::string>> gatherFrom;
     std::optional<uint32_t> demultiplexCount;
```

Comment:
Can you check if making those fields const create any issues? It was initially here to mock gather/demultiplex functionalities in child classes but now those should be created in constructor and can be consts.